### PR TITLE
cert expiration test-case with katello-cert-check

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
### Test Result

```
Testing started at 8:30 PM ...
/home/okhatavk/okhatavk/Satellite/robottelo/env/bin/python /home/okhatavk/Downloads/pycharm-community-2018.1.4/helpers/pycharm/_jb_pytest_runner.py --path /home/okhatavk/okhatavk/Satellite/robottelo/tests/foreman/sys/test_katello_certs_check.py -- -k test_negative_check_expiration_of_certificate
Launching py.test with arguments -k test_negative_check_expiration_of_certificate /home/okhatavk/okhatavk/Satellite/robottelo/tests/foreman/sys/test_katello_certs_check.py in /home/okhatavk/Satellite/robottelo

============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/okhatavk/Satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, cov-2.10.1, xdist-2.2.1, reportportal-5.0.8, mock-3.5.1, forked-1.3.0, ibutsu-1.14.1, picked-0.4.6collected 14 items / 13 deselected / 1 selected

../../okhatavk/Satellite/robottelo/tests/foreman/sys/test_katello_certs_check.py  [100%] 
```

### CODE REVIEW

File: tests/foreman/sys/test_katello_certs_check.py [hunk 1]
```diff
@@ -144,6 +144,14 @@ def generate_certs(self):
             result = connection.run("bash certs.sh")
             assert result.return_code == 0
 
+    @pytest.fixture()
+    def update_system_time(self):
+        result = ssh.command("date -s 'next year'")
+        assert result.return_code == 0
+        yield
+        result = ssh.command("date -s 'last year'")
+        assert result.return_code == 0, 'Failed revert the time setting'
```
Comment (JacobCallahan) [inline tests/foreman/sys/test_katello_certs_check.py:153]:
```suggestion
        assert result.return_code == 0, 'Failed to revert the time setting'
```

File: tests/foreman/sys/test_katello_certs_check.py [hunk 2]
```diff
@@ -144,6 +144,14 @@ def generate_certs(self):
             result = connection.run("bash certs.sh")
             assert result.return_code == 0
 
+    @pytest.fixture()
+    def update_system_time(self):
```
Comment (JacobCallahan) [inline tests/foreman/sys/test_katello_certs_check.py:148]: can you provide a docstring explaining what/why this fixture is used for?

File: tests/foreman/sys/test_katello_certs_check.py [hunk 3]
```diff
@@ -426,22 +434,45 @@ def test_positive_generate_capsule_certs_using_relative_path(
...
+                    cert_data['key_file_name'],
+                    cert_data['ca_bundle_file_name'],
+                ),
+                output_format='plain',
+            )
+        messages = [
+            'Checking expiration of certificate: \n[FAIL]',
+            'Checking CA bundle against the certificate file: \n[FAIL]',
+        ]
+        checks = result.stdout.split("\n\n")
```
Comment (JacobCallahan) [inline tests/foreman/sys/test_katello_certs_check.py:468]:
```suggestion
        checks = result.stdout.split('\n\n')
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Review by jameerpathan111 — APPROVED** [review]
Ack

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**